src/main/java/com/google/firebase/auth/ProviderConfig.java

```diff
@@ -104,4 +104,48 @@ public T setEnabled(boolean enabled) {
 
     abstract T getThis();
   }
+
+  /**
+   * A base class for updating the attributes of an existing provider.
+   */
+  public abstract static class AbstractUpdateRequest<T extends AbstractUpdateRequest<T>> {
+
+    final String providerId;
+    final Map<String,Object> properties = new HashMap<>();
+
+    AbstractUpdateRequest(String providerId) {
+      this.providerId = providerId;
```

Comment:
check not null or empty

Reply:
I already had the check in OidcProviderConfig (so that it could be closer to the Javadoc comment which mentions that it throws an IllegalArgumentException if it's null or empty. But I moved it up here.